### PR TITLE
Fix rowsMax bug

### LINKED ISSUE #1367: Textfield does not follow rowsMax rules on initial render
Reported in FR issue here: https://github.com/equinor/design-system/issues/1336#issuecomment-845973674

### PR DESCRIPTION
**Pull request by pomfrida**
Resolves #1367

### CODE REVIEW

File: libraries/core-react/src/hooks/useAutoResize.ts [hunk 1]
```diff
@@ -14,13 +14,22 @@ export const useAutoResize = (
         newHeight = Math.max(targetEl.scrollHeight, newHeight)
         if (maxHeight) {
```
Comment (mimarz) [inline libraries/core-react/src/hooks/useAutoResize.ts:15]: Is this check necessary as you are already checking it on line 13?

**Review by mimarz — CHANGES_REQUESTED** [review]
Suggest we remove `targetEl.style.height = 'auto'` from the autoReisze hook as its no longer needed?

File: libraries/core-react/src/hooks/useAutoResize.ts (hunk 1 above)
Comment (pomfrida) [reply libraries/core-react/src/hooks/useAutoResize.ts:15]: `targetEl.style.height = 'auto'` is needed here too, for it to autoresize correctly when user deletes text. It's to rewrite the `height` when it's set before.
 
The nested `maxHeight` check is also needed, as you can see its `if` parent is `!maxHeight || maxHeight > newHeight`, this logic is needed for whenever `maxHeight > newHeight` only

Comment (pomfrida) [reply libraries/core-react/src/hooks/useAutoResize.ts]: Improvements fixed and tested so ready again for review  @mimarz

**Review by mimarz — APPROVED** [review]
LGTM 👍

**Pull request merged**